.github/workflows/zenodo-cache-sync.yml

```diff
@@ -0,0 +1,90 @@
+---
+name: zenodo-cache-sync
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - zenodo-cache-action # For testing purposes
```

Comment:
Does it make more sense for this action to run on a push to dev instead of on a schedule? @zaneselvans

Reply:
Well,  it's all either uploading data to GCS, or rsync-ing data between different GCS buckets, so there shouldn't be any cost associated with it running right? That does seem like it would be better for usability. Like as soon as you've got stuff working with the new archive locally, you make a PR, and as soon as that PR with the new archive DOIs in it shows up on `dev` it's available in the cache for CI, the nightly builds, etc.

If it's set run on push to dev, does it also get run when someone makes (or finally merges) a PR into dev? Would it make sense to have it run on `pull_request` and require the sync to finish before the tests run, so that when you're integrating a new DOI / archive all the CI that happens before the new DOI makes it to `dev` can still rely on the zenodo cache?

Reply:
There is no cost associated with running the action.

IIRC, if it's set to run on push to dev, it will run when someone merges a PR into dev. I think it makes sense for this to run before the ci tests run. What is the best way to do this? Use [`workflow_run`](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_run)?

Reply:
Oh ok, so are you saying this action should run on any pull request?

Reply:
I think the only reason we'd want to defer the sync until the changes hit `dev` is if a PR adds a DOI, but it doesn't get merged into `dev` then we'll have some unused archives in the GCS caches. We can always go in and delete them.

Reply:
My experience has been that's pretty rare, so I think not deferring sounds good.

Reply:
Hmm `workflow_run` doesn't seem to be working :/ I also wonder if running `zenodo-cache-sync` on push and pull_request is be problematic because we'll have two runners writing and syncing to the GCS buckets.